### PR TITLE
Port config and launch to ROS 2

### PR DESCRIPTION
**Pull request by cottsay**
Also fix the header's install location.

Note that the ROS 1 version of this package could handle a bare value in place of the 3-part map, but that doesn't appear to be the case in the ROS 2 version. I adjusted the configs to accommodate that.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,11 +39,11 @@ install(TARGETS ${PROJECT_NAME}
 install(TARGETS ${PROJECT_NAME}_node
   DESTINATION lib/${PROJECT_NAME})
 
-install(DIRECTORY include/${PROJECT_NAME}/
+install(DIRECTORY include/
```
Comment (clalancette) [inline CMakeLists.txt:42]: Good catch.

File: launch/teleop.launch.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from ament_index_python.packages import get_package_share_directory
+
+import launch
+import launch_ros.actions
+
+
+def generate_launch_description():
+    joy_config = launch.substitutions.LaunchConfiguration('joy_config')
+    joy_dev = launch.substitutions.LaunchConfiguration('joy_dev')
+    config_filepath = launch.substitutions.LaunchConfiguration('config_filepath')
```
Comment (clalancette) [inline launch/teleop.launch.py]: This is my first experience with `launch.Substitutions`, so this is a question: what does this do?  Also, do we need to both declare the substitutions up here and then use `DeclareLaunchArgument` below?  It seems a bit redundant.

File: package.xml [hunk 3]
```diff
@@ -15,6 +15,7 @@
   <depend>geometry_msgs</depend>
   <depend>rclcpp</depend>
   <depend>sensor_msgs</depend>
+  <exec_depend>joy</exec_depend>
```
Comment (clalancette) [inline package.xml:19]: Nit: space before this line.

**Review by clalancette — CHANGES_REQUESTED** [review]
I've got one nit, and one question for clarification.  Other than that, looks good to me.

File: launch/teleop.launch.py (hunk 2 above)
Comment (cottsay) [reply launch/teleop.launch.py]: This answers post by William does a pretty good job describing the difference: http://answers.ros.org/question/322874/ros2-what-is-different-between-declarelaunchargument-and-launchconfiguration/?answer=322877#post-id-322877

For better or worse, I think if `launch.Substitututions` as a mechanism for delayed expansion of a variable, and `DeclareLaunchArgument` as one possible mechanism for actually changing the value of that variable.

File: launch/teleop.launch.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import os
```
Comment (clalancette) [inline launch/teleop.launch.py]: One nit; I think we've determined that for tab-completion, `teleop-launch.py` is better.

**Review by clalancette — CHANGES_REQUESTED** [review]
One more nit here, then this is good to go in.

File: launch/teleop.launch.py (hunk 4 above)
Comment (cottsay) [reply launch/teleop.launch.py]: Done!

**Review by clalancette — APPROVED** [review]
The Eloquent CI build failed, but that's because joystick_drivers hasn't been released there yet.  I'm happy with taking the Dashing as a passing grade, so I'll approve and merge this.  Thanks!

**Pull request merged**